contrib/mesos/pkg/minion/tasks/task.go

```diff
@@ -240,7 +247,8 @@ type exitError interface {
 	Sys() interface{}
 }
 
-func taskRunning(t *Task) taskStateFn {
+func taskRunning(t *Task) (nextState taskStateFn) {
```

Comment:
You can parametrize the state fn. then the code exists only once. But the state machine via the for-loop inside a state machine is superfluous IMO. Just create a taskTerminating with force argument. 

> Am 24.09.2015 um 16:27 schrieb James DeFelice <notifications@github.com>:
> 
> In contrib/mesos/pkg/minion/tasks/task.go:
> 
> > @@ -240,7 +247,8 @@ type exitError interface {
> >  	Sys() interface{}
> >  }
> >  
> > -func taskRunning(t *Task) taskStateFn {
> > +func taskRunning(t *Task) (nextState taskStateFn) {
> This seemed cleaner, more DRY. Otherwise basically the same code blob gets copy-pasted elsewhere. I'm happy to refactor the loop into a separate func that's more easily testable.
> 
> —
> Reply to this email directly or view it on GitHub.
>